crates/flowctl/src/config.rs

```diff
@@ -1,29 +1,114 @@
 use anyhow::Context;
-use serde::{Deserialize, Serialize};
 use std::path::PathBuf;
 
-lazy_static::lazy_static! {
-    static ref DEFAULT_DASHBOARD_URL: url::Url = url::Url::parse("https://dashboard.estuary.dev/").unwrap();
-}
-
-#[derive(Debug, Serialize, Deserialize, Default)]
+/// Configuration of `flowctl`.
+///
+/// We generally keep this minimal and prefer to use built-in default
+/// or local value fallbacks, because that means we can update these
+/// defaults in future releases of flowctl without breaking local
+/// User configuration.
+#[derive(Debug, Default, serde::Serialize, serde::Deserialize)]
 pub struct Config {
+    /// URL endpoint of the Flow control-plane Agent API.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub agent_url: Option<url::Url>,
     /// URL of the Flow UI, which will be used as a base when flowctl generates links to it.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
     pub dashboard_url: Option<url::Url>,
     /// ID of the current draft, or None if no draft is configured.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
     pub draft: Option<models::Id>,
-    // Current access token, or None if no token is set.
-    pub api: Option<API>,
+    /// Public (shared) anonymous token of the control-plane API.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub pg_public_token: Option<String>,
+    /// URL endpoint of the Flow control-plane PostgREST API.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub pg_url: Option<url::Url>,
+    /// Users's access token for the control-plane API.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub user_access_token: Option<String>,
+    /// User's refresh token for the control-plane API,
+    /// used to generate access_token when it's unset or expires.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub user_refresh_token: Option<RefreshToken>,
+
+    #[serde(skip)]
+    is_local: bool,
```

Comment:
optional, just tossing out the idea: Instead of all the matches in the `get_*` functions, it might be a bit cleaner to have a struct like

```
struct ProfileDefaults {
  agent_url: Url,
  dashboard_url: Url,
  ...
}
```

and 

```
#[serde(skip)]
profile_defaults: ProfileDefaults
```